### PR TITLE
Add paper and preprint section to `index.md`

### PR DESCRIPTION
**Pull request by jbisits**
Can be added somewhere else if you think better? 

I started using the the vs-code settings as suggested by the [SciML Style Guide for Julia](https://github.com/SciML/SciMLStyle) (towards the bottom of the `README`) which gets rid of extra spaces at the end of lines and extraneous extra lines. This makes it looks like there are a fair few changes even though all I have added is the paper section to `index.md`. Sorry if this is annoying!

Once the paper is released (not just early online release) I will update the entry.

### CODE REVIEW

**Comment by jbisits** [comment]
Will have to look into what is happening with the tests some other time this week sorry! No real rush on this one.

**Comment by jbisits** [comment]
I merged my fork upstream, maybe something went wrong with that or I have some other changes left here that I did not correct.

File: docs/src/index.md [hunk 1]
```diff
@@ -81,3 +76,7 @@ The bibtex entry for the paper is:
   journal = {Journal of Open Source Software}
 }
 ```
+
+## Papers and preprints using `GeophysicalFlows.jl`
+
+1. Bisits, J. I., Stanley G. J., Zika, J. D. Can we accurately quantify a lateral diffusivity using a single tracer release? _Journal of Physical Oceanography_, Early Online Release, DOI: 10.1175/JPO-D-22-0145.1
```
Comment (navidcy) [inline docs/src/index.md:82]:
```suggestion
1. Bisits, J. I., Stanley G. J., and Zika, J. D. (2022). Can we accurately quantify a lateral diffusivity using a single tracer release? _Journal of Physical Oceanography_, Early Online Release, doi:[10.1175/JPO-D-22-0145.1](https://doi.org/10.1175/JPO-D-22-0145.1)
```

**Comment by navidcy** [comment]
@thdrivas, can you write here for us any papers of yours (or others that you know) that used GeophysicalFlows.jl?

**Comment by navidcy** [comment]
@jbisits can we keep this PR only for the `index.md`? let's do a different one for code beautification?

**Comment by navidcy** [comment]
(perhaps you can remove some commits with `git rebase -i` (I've never done it -- google it?) :)

File: test/runtests.jl [hunk 2]
```diff
@@ -98,12 +98,12 @@ for dev in devices
     @test test_bqgql_advection(0.0005, "ForwardEuler", dev)
     @test test_bqgql_energyenstrophy(dev)
     @test test_bqgql_problemtype(dev, Float32)
-    @test BarotropicQGQL.nothingfunction() == nothing
+    @test isnohting(BarotropicQGQL.nothingfunction())
```
Comment (navidcy) [inline test/runtests.jl:101]: typo here

but I'd urge to keep the PR to its title?

**Comment by jbisits** [comment]
> @jbisits can we keep this PR only for the `index.md`? let's do a different one for code beautification?

Yes good idea. I will revert these changes.

File: test/runtests.jl (hunk 2 above)
Comment (jbisits) [reply test/runtests.jl:101]: Yep, will do.

**Comment by navidcy** [comment]
Closing in favor of #309

**Pull request closed without merging**